arangod/GeneralServer/AcceptorTcp.cpp

```diff
@@ -151,7 +152,10 @@ void AcceptorTcp<SocketType::Tcp>::asyncAccept() {
   // _asioSocket is nullptr.
   TRI_ASSERT(_endpoint->encryption() == Endpoint::EncryptionType::NONE);
 
-  _asioSocket.reset(new AsioSocket<SocketType::Tcp>(_server.selectIoContext()));
+  {
+    std::unique_lock<std::mutex> guard(_mutex);
+    _asioSocket = std::make_unique<AsioSocket<SocketType::Tcp>>(_server.selectIoContext());
```

Comment:
In line 192 we still access `_asioSocket` without the mutex and read its `socket` and `peer` members. I think this is safe, because even though `_asioSocket` is initialized after line 157, no other thread can call `close()` on it and reset the `_asioSocket` member.
But it is a bit tricky to figure all this out.

I think keeping the `_asioSocket` member (in contrast to passing the asio socket in local variables) still makes sense, because I don't see a good other way of how `close()` would work.

Reply:
As discussed offline, all the issues related to `_asioSocket` can be avoided by completely removing that member since it is not required. The timer is only initialized once the the CommTask has been created, but at that point the `_asioSocket` has been moved from the acceptor into the CommTask anyway, so there is no point in cancelling the timer in `close()`.